### ISSUE TITLE
Ratchet And Clank Future: Tools of Destruction Freezes randomly

### ISSUE THREAD

**Issue by GioHiTech** [RPCS3/rpcs3#8653]
Been trying to play the game but every time the game freezes on a random part of the level but keeps playing audio. Here are my specs:

Ryzen 5 2400G
GTX 1060 6GB
16GB RAM

Would there be any known fix to this? I'm aware the processor is relatively weak for emulation, the framerates tank sometimes but I'd be willing to play the game if there is a way to play the game without these random crashes, or at least be able to finish the first level.

**Comment by GioHiTech** [comment]
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/4997167/RPCS3.log.gz)

**Comment by isJuhn** [comment]
are you using [patch](https://wiki.rpcs3.net/index.php?title=Help:Game_Patches#Ratchet_.26_Clank:_Tools_of_Destruction)?

**Comment by GioHiTech** [comment]
No, I wasn't aware of it. How would I go about applying these patches?

**Comment by isJuhn** [comment]
it says at the top of the page

**Comment by GioHiTech** [comment]
The patches worked for ToD and QFB, but A Crack in Time still suffers from this problem. I've also noticed that there's a big chance that the game will crash with a "Memory Allocation Underflow! x44" whenever I toss the wrench at an enemy or a crate.

**Comment by arcadee1977** [comment]
> are you using patch?

Why remove this patch now?

**Comment by isJuhn** [comment]
@arcadee1977 it seems only game hacks and enhancements are allowed on the main page, patches that workaround emulator inaccuracies has been moved to the [Canary](https://wiki.rpcs3.net/index.php?title=Help:Game_Patches/Canary) page.

@GioHiTech added ACiT freeze fix, make sure to use version 1.00 as 1.20 has SPU freezes that I haven't figured out.

**Comment by arcadee1977** [comment]
thanks@isJuhn ,but you also have ACiT:BCAS20098 or NPEA00453 's freeze fix patch?

**Comment by nitroos-ai** [comment]
Is this still an issue?

**Comment by DoudsonFire** [comment]
Yeah Having the same issue at axiom city

**Comment by titoshadow** [comment]
Yes, its still an issue, it freezes randomly

**Comment by kd-11** [comment]
Try https://github.com/RPCS3/rpcs3/issues/12585

**Comment by titoshadow** [comment]
https://github.com/RPCS3/rpcs3/pull/13859 this ?

**Comment by kd-11** [comment]
> #13859 this ?

Indeed, yes. In low memory scenarios crashes can still happen, but maybe not as frequently.

**Comment by titoshadow** [comment]
Well I have been able to play quite longer, but still having issues (repeatable, recurrent in the same exact point) when fighting the batallaplex hydra, in which the game freezes after 3 seconds.

Will try to get more info on this, since it seems to be a different kind of animal.

Thanks for your help !

**Comment by GioHiTech** [comment]
Coming back to this issue after a while, ACiT runs much better, however the game (more often than not) crashes when I throw Ratchet's wrench at something, and tethering onto things (which is essential to progression) also counts. I have the canary patch applied on BCUS98124 version 1.00.

Ryzen 5600x / GTX 1060 6GB

**Comment by GioHiTech** [comment]
Here's a crash report from when I threw a wrench in Torren IV (I had the issue many times on Vorselon's ship but managed to, somehow, not crash the game and keep progressing)

[RPCS3.gz](https://github.com/RPCS3/rpcs3/files/12325317/RPCS3.gz)

**Comment by GioHiTech** [comment]
Okay, this is going to sound ridiculous, but: I'm playing the game on keyboard and I bound the R2 (ready throw) key to left CTRL. Rebinding it to a mouse button just... fixed the issue, I don't know why.

**Comment by titoshadow** [comment]
@GioHiTech do you mean the game is not crashing anymore ?

**Comment by GioHiTech** [comment]
@titoshadow for me, nope. After I got the issue fixed I had like 2 crashes across a whole playthrough

**Comment by digant73** [comment]
@titoshadow, @GioHiTech this issue was possibly fixed. Please retest when possible

**Comment by alexconsole04** [comment]
It is still an issue across all the ps3 Ratchet and clank games.

**Comment by kd-11** [comment]
Confirming as well. I'll post a log later, but I had this happen to me as well.

**Comment by Proutbedaine** [comment]
I just posted here about further issue of the sort: https://github.com/RPCS3/rpcs3/issues/15289#issuecomment-5040845940

The game kept freezing as soon as anything new would compile (so I kept going sliiiightly further in the intro of the game when you start playing as clank, but as soon as it had to compile it froze and audio would continue and I had to terminate the process each time.)
(Whether I played with all the patches ON or OFF, same whether I was on v1.00 or v1.20 of the game. Whether I played with the default configs ON or custom configs. And tons of different permutations of custom settings. Always got a tiny bit further than last time and then it would freeze again.)

Setting Shader Mode to "Async Recompiler (multi-threaded)" instead of "Async Recompiler with Shader Interpreter" was the only thing that solved it so far.

More details in logs in the comment on the other issue thread.

**Comment by kaixoo12** [comment]
I'm not entirely sure if this is the same issue, because I'm incompetent at reading log files. I reported a crash when starting the first level playing as Clank after having applied the Freeze fix. A user from Discord recommended me to switch "<kbd>GPU</kbd> -> <kbd>Shader Mode</kbd>" to <kbd>Async Recompiler (multi-threaded)</kbd> and, after applying this setting, I get no more crashes.

Might it be related to this issue? If it's not I'll create a new issue but I wanted to check first.

P.S: It was the <kbd>Async Recompiler (interpreter)</kbd> mode that was causing the crash

My log:

[RPCS3.log](https://github.com/user-attachments/files/30284006/RPCS3.log)
[RPCS3.log.gz](https://github.com/user-attachments/files/30284007/RPCS3.log.gz)

**Comment by Proutbedaine** [comment]
> I'm not entirely sure if this is the same issue, because I'm incompetent at reading log files. I reported a crash when starting the first level playing as Clank after having applied the Freeze fix. A user from Discord recommended me to switch "GPU -> Shader Mode" to Async Recompiler (multi-threaded) and, after applying this setting, I get no more crashes.
> 
> Might it be related to this issue? If it's not I'll create a new issue but I wanted to check first.
> 
> P.S: It was the Async Recompiler (interpreter) mode that was causing the crash
> 
> My log:
> 
> [RPCS3.log](https://github.com/user-attachments/files/30284006/RPCS3.log) [RPCS3.log.gz](https://github.com/user-attachments/files/30284007/RPCS3.log.gz)

Yeah it seems related to this one here: https://github.com/RPCS3/rpcs3/issues/15289#issuecomment-5040845940

Which seems to be related to when any option that has the "Shader Interpreter" function in it that makes the game crash as soon as there's compiling needed to be done; seemingly always through a "VK_ERROR_DEVICE_LOST" type error.

And the only fix for those crash is to simply use only the "Async Recompiler (multi-threaded)" option instead.

**Comment by kd-11** [comment]
I'll add clarification here - this ticket is for the CPU-side freezing. This game has a CPU-side freezing issue that is separate from the GPU-side freezing issue caused by the driver crashing during shader interpreter execution.
The GPU side crash is tracked in https://github.com/RPCS3/rpcs3/issues/19073
So if you're having issues that go away when the shader compiler is set to async mode (without interpreter) that other ticket will keep track of it. This ticket should remain for the much older ticket where the game logic itself freezes but the GPU is still working fine. You can open the home menu and exit for example which is impossible if the GPU had crashed.

PS: I recently completed the game on an older version of the emulator and it is still unplayable because of the hangs. Especially in the battleplex it is infuriating to have the logic sieze up in round 18 of 20. In one case it was round 20 of 20 and froze.
